CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.15)
+
+set(PROJECT_NAME template-project)
+set(PROJECT_VERSION 1.0.0)
```

Comment:
no need to define specific variable for project_version. 

project(${PROJECT_NAME}
        VERSION 1.0.0
        LANGUAGES CXX)

after this definition, you're able to access version via PROJECT_VERSION. (https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION.html)

same is also applicable for PROJECT_NAME